src/Maestro/Maestro.Web/Api/v2020_02_20/Controllers/ChannelsController.cs

```diff
@@ -0,0 +1,242 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Maestro.Data;
+using Maestro.Data.Models;
+using Microsoft.AspNetCore.ApiVersioning;
+using Microsoft.AspNetCore.ApiVersioning.Swashbuckle;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Build = Maestro.Data.Models.Build;
+using Channel = Maestro.Web.Api.v2020_02_20.Models.Channel;
+using FlowGraph = Maestro.Web.Api.v2018_07_16.Models.FlowGraph;
+
+namespace Maestro.Web.Api.v2020_02_20.Controllers
+{
+    /// <summary>
+    ///   Exposes methods to Create/Read/Edit/Delete <see cref="Channel"/>s.
+    /// </summary>
+    [Route("channels")]
+    [ApiVersion("2020-02-20")]
+    public class ChannelsController : v2018_07_16.Controllers.ChannelsController
+    {
+        private readonly BuildAssetRegistryContext _context;
+        private readonly IRemoteFactory _remoteFactory;
+
+        public ChannelsController(BuildAssetRegistryContext context,
+                                  IRemoteFactory factory,
+                                  ILogger<ChannelsController> logger)
+           : base(context, factory, logger)
+        {
+            _context = context;
+            _remoteFactory = factory;
+        }
+
+        /// <summary>
+        ///   Gets a list of all <see cref="Channel"/>s that match the given classification.
+        /// </summary>
+        /// <param name="classification">The <see cref="Channel.Classification"/> of <see cref="Channel"/> to get</param>
+        [HttpGet]
+        [SwaggerApiResponse(HttpStatusCode.OK, Type = typeof(List<Channel>), Description = "The list of Channels")]
+        [ValidateModelState]
+        public override IActionResult ListChannels(string classification = null)
```

Comment:
Why is this code here? It's a duplicate of the underlying method?

Reply:
Ooo. 'Channel' is different. That makes sense.  It seems like we could do better code sharing here (since the only things that's different is the part that transforms the internal model into the API model for a lot of these).

---
In reply to: [385936779](https://github.com/dotnet/arcade-services/pull/966#discussion_r385936779) [](ancestors = 385936779)